### PR TITLE
Added code coverage

### PR DESCRIPTION
**Pull request by KjellKod**
Added code coverage
Added unique build number from git commit numbers
library is shared
library will have unique number

screen shot 2016-01-05 at 2 53 50 pm

### CODE REVIEW

File: scripts/codeCoverage.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+LAUNCH_DIR=`pwd`
+COVERAGE_DIR=coverage
+SOURCE_BLACKLIST=".*ProbeTransmogrifier.cpp.*"
```
Comment (yannick-polius) [inline scripts/codeCoverage.sh]: old probet reference

File: scripts/codeCoverage.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+HEADER_WHITELIST=
+
+#clean up coverage dir
+rm -rf $COVERAGE_DIR
+mkdir -p $COVERAGE_DIR
+
+PROJECT="ProbeTransmogrifier"
```
Comment (yannick-polius) [inline scripts/codeCoverage.sh]: old probet reference

File: scripts/codeCoverage.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+OBJECT_DIR="probe_transmogrifier_debug.dir"
```
Comment (yannick-polius) [inline scripts/codeCoverage.sh]: old probet reference

**Comment by yannick-polius** [comment]
:+1:  after removing old probet references

File: scripts/codeCoverage.sh (hunk 1 above)
Comment (KjellKod) [inline scripts/codeCoverage.sh]: done

File: scripts/codeCoverage.sh (hunk 2 above)
Comment (KjellKod) [inline scripts/codeCoverage.sh]: tx. Done

**Comment by KjellKod** [comment]
@yannick-polius and @shwetabhandare please re-review

**Comment by yannick-polius** [comment]
:+1:

**Comment by KjellKod** [comment]
@KjellKod make sure that code coverage works if the rpm build and tests are executed as root user

File: packaging/DeathKnell.spec [hunk 4]
```diff
@@ -28,13 +28,25 @@ rm -f  CMakeCache.txt
...
+
+make -j6
 ./UnitTestRunner
+
+
 mkdir -p $RPM_BUILD_ROOT/usr/local/probe/lib
-cp *.so $RPM_BUILD_ROOT/usr/local/probe/lib
-rm $RPM_BUILD_ROOT/usr/local/probe/lib/libgtest_170_lib.so
+cp -rfd lib%{name}.so* $RPM_BUILD_ROOT/usr/local/probe/lib
+rm -f RPM_BUILD_ROOT/usr/local/probe/lib/libgtest_170_lib.so
```
Comment (john-gress) [inline packaging/DeathKnell.spec]: You still need the $ sign on RPM_BUILD_ROOT.

Comment (KjellKod) [inline packaging/DeathKnell.spec]: tx.

Comment (john-gress) [inline packaging/DeathKnell.spec]: But with your changes to line 48, do you even need this line anymore?

Comment (KjellKod) [inline packaging/DeathKnell.spec]: I'm updating the other LROSS with the same fix

File: scripts/codeCoverage.sh [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+GIT_VERSION=`git rev-list HEAD --count`
+VERSION="1.$GIT_VERSION"
+
+
+echo "Pseudo FileIO version: $VERSION"
+PATH=/usr/local/probe/bin:$PATH
+/usr/local/probe/bin/cmake -DUSE_LR_DEBUG=ON -DVERSION=$VERSION -DCMAKE_CXX_COMPILER_ARG1:STRING=' -Wall -Werror -g -gdwarf-2 -fprofile-arcs -ftest-coverage -O0 -fPIC -m64 -Wl,-rpath -Wl,. -Wl,-rpath -Wl,/usr/local/probe/lib -Wl,-rpath -Wl,/usr/local/probe/lib64 ' -DCMAKE_CXX_COMPILER=/usr/local/probe/bin/g++ ..
+
+
+make -j
```
Comment (john-gress) [inline scripts/codeCoverage.sh]: Most other places you use -j6.

Comment (KjellKod) [inline scripts/codeCoverage.sh]: fixed

**Comment by craig-cogdill** [comment]
:+1:

**Pull request merged**